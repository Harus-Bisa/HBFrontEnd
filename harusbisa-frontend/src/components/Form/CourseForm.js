import React from "react";
import {connect} from "react-redux";
import { addCourse, editCourse } from "../../redux/actions";
import { Form, FormGroup, Label, Input, FormFeedback, FormText, Col} from 'reactstrap';
import { HB_YELLOW } from "../../css/constants/color";
import { Button } from "@material-ui/core";

function mapStateToProps(state, ownProps){
    if (ownProps.id){
        let courses = state.courses
        for (let i=0; i<courses.length; i++){
            if (courses[i]._id === ownProps.id){
                return{
                    course: courses[i]
                }
            }
        }
        return null
    }
    else{
        return {
            course:{
                course_name: "",
                start_term: "",
                end_term: ""
            }
        }
    }
    
}
function CourseForm(props){
    const [name, setName] = React.useState(props.course.course_name);
    const [startDate, setStartDate] = React.useState(props.course.start_term);
    const [endDate, setEndDate] = React.useState(props.course.end_term);
    
    const submit = (event) =>{
        event.preventDefault();
        if (props.id){
            props.editCourse(props.id, name, startDate, endDate);
        }
        else{
            props.addCourse(name, startDate, endDate);
        }
        props.closePopup();
    }
    return(
        <div className="container-fluid">
            {/* <form onSubmit={submit}>
                <label>Course Name*</label>
                <input type="text" value={name} onChange={(event) => setName(event.target.value)}></input>
                <br/>
                <label>Start Date*</label>
                <input type="text" value={startDate} onChange={(event) => setStartDate(event.target.value)}></input>
                <br/>
                <label>End Date*</label>
                <input type="text" value={endDate} onChange={(event) => setEndDate(event.target.value)}></input>
                <br/>
                <button type="submit">Submit</button>
            </form> */}
            <div className="row" style={{borderBottom:'2px solid '+HB_YELLOW}}>
                <div className="col-12" style={{display:'flex', justifyContent:'center', padding:'1rem 2rem'}}>
                    <h3>Tambahkah Kelas Baru</h3>
                </div>
            </div>
            <div className="content">
                <Form onSubmit={submit}>
                    <FormGroup row>
                        <Label sm={5}>Nama Kelas</Label>
                        <Col sm={7}>
                            <Input type="text" value={name} placeholder={"Contoh: Biologi Molekuler Kelas A"} onChange={(event) => setName(event.target.value)}/>
                            <FormText>Nama ini adalah nama yang akan dilihat siswa Anda.</FormText>
                        </Col>
                    </FormGroup>
                    <FormGroup row>
                        <Label sm={5}>Periode Kelas</Label>
                        <Col sm={7}>
                            <div className="row">
                                <div className="col-5">
                                    <Input type="select" name="select" id="exampleSelect">
                                        <option>Juli 2019</option>
                                        <option>Juli 2020</option>
                                        <option>Juli 2021</option>
                                        <option>Juli 2022</option>
                                        <option>Juli 2023</option>
                                    </Input>
                                </div>
                                <div className="col-2">
                                    <p>-</p>
                                </div>
                                <div className="col-5">
                                    <Input type="select" name="select" id="exampleSelect">
                                        <option>Juli 2019</option>
                                        <option>Juli 2020</option>
                                        <option>Juli 2021</option>
                                        <option>Juli 2022</option>
                                        <option>Juli 2023</option>
                                    </Input>
                                </div>
                            </div>
                        </Col>
                    </FormGroup>
                    <Button className="prof-button" type="submit">Submit</Button>
                </Form>
            </div>
        </div>
    )
}

export default connect(mapStateToProps,{addCourse, editCourse})(CourseForm);