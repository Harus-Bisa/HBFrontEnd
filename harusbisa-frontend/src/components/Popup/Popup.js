import React from "react";
import { Dialog, DialogContent, Button } from "@material-ui/core";
import CourseForm from "../Form/CourseForm";
import { removeError } from "../../redux/actions";
import {connect} from "react-redux";

function Popup(props){
    const [open, setOpen] = React.useState(false);
    const ContentComponent = props.content
    const Trigger = props.trigger.component
    
    const openPopup = () =>{
        setOpen(true)
        props.removeError()
    }
    return(
        <div>
            <Trigger className={props.trigger.className} onClick={openPopup}>{props.purpose}</Trigger>
            <Dialog open={open} onClose={() => setOpen(false)} style={{padding:0}}>
                <DialogContent> 
                    <ContentComponent closePopup={() => setOpen(false)} id={props.id}/>
                </DialogContent>
            </Dialog>
        </div>
    )   
}

Popup.defaultProps={
    purpose: "default purpose",
    trigger:{
        component: Button
    },
    content: CourseForm
}

export default connect(null, {removeError})(Popup)