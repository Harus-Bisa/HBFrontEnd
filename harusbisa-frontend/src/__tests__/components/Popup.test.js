import React from "react";
import TestRenderer  from "react-test-renderer";
import {shallow} from "enzyme";
import '../../test-config';
import Popup from "../../components/Popup/Popup";
import { Button, Dialog } from "@material-ui/core";
import CourseForm from "../../components/Form/CourseForm";

describe("StudentCourseCard component", () =>{

    it("Matches snapshot", () =>{
        let testRenderer = TestRenderer.create(<Popup/>)
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })

    it("opens dialog", () =>{
        let wrapper = shallow(<Popup/>)
        var trigger = wrapper.find(Button)
        var dialog = wrapper.find(Dialog)
        var content = wrapper.find(CourseForm)

        expect(trigger).toBeDefined()
        expect(dialog).toBeDefined()
        expect(dialog.props().open).toBeFalsy()
        expect(content).toBeDefined()

        trigger.simulate('click')
        expect(wrapper.find(Dialog).props().open).toBeTruthy()
        wrapper.find(Dialog).simulate('close')
        expect(wrapper.find(Dialog).props().open).toBeFalsy() 
    })
})