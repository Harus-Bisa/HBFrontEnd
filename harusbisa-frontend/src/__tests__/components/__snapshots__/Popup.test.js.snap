// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component Matches snapshot 1`] = `"<div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">default purpose</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>"`;
